src/jemalloc.c

```diff
@@ -1619,6 +1624,34 @@ malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
 				}
 				CONF_CONTINUE;
 			}
+			if (config_uaf_detection &&
+			    CONF_MATCH("lg_san_uaf_align")) {
+				ssize_t a;
+				CONF_VALUE_READ(ssize_t, a)
+				if (CONF_VALUE_READ_FAIL() || a < -1) {
+					CONF_ERROR("Invalid conf value",
+					    k, klen, v, vlen);
+				}
+				/* TODO: add init function after rebase. */
+				if (a == -1) {
+					opt_lg_san_uaf_align = -1;
+					cache_bin_nonfast_mask = (uintptr_t)-1;
+					CONF_CONTINUE;
+				}
+
+				/* clip if necessary */
+				ssize_t max_allowed = (sizeof(size_t) << 3) - 1;
```

Comment:
Agreed it's easier to read that way. However I'm a bit hesitant since we don't have any use of CHAR_BIT currently, and we are doing an upstream release very soon -- we do include limits.h but I'm a bit worried about things being different on some versions of MSVC etc.

Since we have a few other cases in this function also doing the `<< 3`, how about we keep it for now?